saw-central/src/SAWCentral/Crucible/LLVM/Builtins.hs

```diff
@@ -981,72 +981,6 @@ verifyPrestate opts cc mspec globals =
-  MS.CrucibleMethodSpecIR (LLVM arch) ->
-  ProofScript () ->
-  [Crucible.LabeledPred Term AssumptionReason] ->
-  TopLevel Bool
-assumptionsContainContradiction cc methodSpec tactic assumptions =
-  do
-     let sym = cc^.ccSym
-     st <- io $ Common.sawCoreState sym
-     let sc  = saw_ctx st
-     let ploc = methodSpec^.MS.csLoc
-     (goal',pgl) <- io $
-      do
-         -- conjunction of all assumptions
-         assume <- scAndList sc (toListOf (folded . Crucible.labeledPred) assumptions)
-         -- implies falsehood
-         goal  <- scImplies sc assume =<< toSC sym st (W4.falsePred sym)
-         goal' <- boolToProp sc [] goal
-         return $ (goal',
-                  ProofGoal
-                  { goalNum  = 0
-                  , goalType = "vacuousness check"
-                  , goalName = show (methodSpec^.MS.csMethod)
-                  , goalLoc  = show (W4.plSourceLoc ploc) ++ " in " ++ show (W4.plFunction ploc)
-                  , goalDesc = "vacuousness check"
-                  , goalSequent = propToSequent goal'
-                  , goalTags = mempty
-                  })
-     res <- runProofScript tactic goal' pgl Nothing
-              "vacuousness check" False False
-     case res of
-       ValidProof _ _     -> return True
-       InvalidProof _ _ _ -> return False
-       UnfinishedProof _  ->
-         -- TODO? is this the right behavior?
-         do printOutLnTop Warn "Could not determine if preconditions are vacuous"
-            return True
-
--- | Given a list of assumptions, computes and displays a smallest subset of
--- them that are contradictory among each themselves.  This is **not**
--- implemented efficiently.
-computeMinimalContradictingCore ::
-  (Crucible.HasPtrWidth (Crucible.ArchWidth arch), Crucible.HasLLVMAnn Sym) =>
-  LLVMCrucibleContext arch ->
-  MS.CrucibleMethodSpecIR (LLVM arch) ->
-  ProofScript () ->
-  [Crucible.LabeledPred Term AssumptionReason] ->
-  TopLevel ()
-computeMinimalContradictingCore cc methodSpec tactic assumes =
-  do
-     printOutLnTop Warn "Contradiction detected! Computing minimal core of contradictory assumptions:"
-     -- test subsets of assumptions of increasing sizes until we find a
-     -- contradictory one
-     let cores = sortBy (compare `on` length) (subsequences assumes)
-     findM (assumptionsContainContradiction cc methodSpec tactic) cores >>= \case
-      Nothing -> printOutLnTop Warn "No minimal core: the assumptions did not contains a contradiction."
-      Just core ->
-        forM_ core $ \assume ->
-          case assume^.Crucible.labeledPredMsg of
-            (loc, reason) -> printOutLnTop Warn (show loc ++ ": " ++ reason)
-     printOutLnTop Warn "Because of the contradiction, the following proofs may be vacuous."
-
 -- | Checks whether the given list of assumptions contains a contradiction, and
 -- if so, computes and displays a minimal set of contradictory assumptions.
 checkAssumptionsForContradictions ::
```

Comment:
Now that `checkAssumptionsForContradictions` has been moved to `SAWCentral.Crucible.Common.Vacuity`, I would favor removing this function here, as it is essentially a duplicate.